instctrl.py

```diff
@@ -48,4 +49,17 @@ def PUT(self, name):
     @json_response
     @api_response
     def DELETE(self, name):
-        raise Error(NOTALLOWED)
+        """Deletes institution using institution uuid."""
+        data = json.loads(web.data())
+        uuid = data.get('institution_uuid')
+        try:
+            assert uuid
+        except AssertionError as error:
+            logger.debug(error)
+            raise Error(BADPARAMS)
+        results = Institution.get_institution(uuid)
+        for result in results:
+            institution_to_be_deleted = Institution(result['institution_uuid'],
+                 result['institution_name'],result['institution_country_code'])
+            institution_to_be_deleted.delete()
+        return results_to_institutions(results) # TODO why doesn't it return institutions that are deleted?
```

Comment:
results_to_institutions() simply instantiates the object, which you already have, so there is no need to call it - you can just do
`
try:
    result = Institution.get_institution(uuid)[0]
except:
    raise Error(NOTFOUND, msg="The institution provided does not exist.")

institution = Institution(result['institution_uuid'], result['institution_name'],result['institution_country_code'])
institution.delete()
return [institution.__dict__]
`